### PR TITLE
[Testing helpers][Diagnostics]: Add namespace checks.

### PR DESCRIPTION
**Pull request by amanda-tarafa**

### CODE REVIEW

File: tools/Google.Cloud.ClientTesting/CodeHealthTester.cs [hunk 1]
```diff
@@ -67,5 +67,33 @@ public static void AssertClassesAreSealedOrAbstract(Type sampleType, IEnumerable
             // Force output to show the bad types
             Assert.Equal(new string[0], badTypes.ToList());
         }
+
+        /// <summary>
+        /// Asserts that the assembly containing the given type does not define the disallowed namespaces.
+        /// </summary>
+        public static void AssertNoDisallowedNamespaces(Type sampleType, IEnumerable<string> disallowedNamespeces)
```
Comment (jskeet) [inline tools/Google.Cloud.ClientTesting/CodeHealthTester.cs:74]: If you make this a `params` array, you won't need the `new string[]` in the calling code.

File: tools/Google.Cloud.ClientTesting/CodeHealthTester.cs [hunk 2]
```diff
@@ -67,5 +67,33 @@ public static void AssertClassesAreSealedOrAbstract(Type sampleType, IEnumerable
...
+        {
+            var badNamespaces = sampleType.Assembly.DefinedTypes
+                .Select(t => t.Namespace)
+                .Distinct()
+                .Intersect(disallowedNamespeces);
+
+            // Force output to show the bad namespaces.
+            Assert.Equal(new string[0], badNamespaces.ToList());
```
Comment (jskeet) [inline tools/Google.Cloud.ClientTesting/CodeHealthTester.cs:82]: I think `Assert.Empty(badNamespaces)` would show everything, wouldn't it

I think I'd be tempted to list the types themselves - so use `sampleType.Assembly.DefinedTypes.Where(t => disallowedNamespaces.Contains(t.Namespace))` and the equivalent for AssertOnlyAllowed.

File: tools/Google.Cloud.ClientTesting/CodeHealthTester.cs (hunk 1 above)
Comment (jskeet) [inline tools/Google.Cloud.ClientTesting/CodeHealthTester.cs:74]: Typo nit (here and below) - namespaces, not namespeces

File: tools/Google.Cloud.ClientTesting/CodeHealthTester.cs (hunk 2 above)
Comment (amanda-tarafa) [reply tools/Google.Cloud.ClientTesting/CodeHealthTester.cs]: Done.

**Review by amanda-tarafa — COMMENTED** [review]
Comments addressed in original commits because it's all there is.

**Review by jskeet — APPROVED** [review]

**Pull request merged**